jerry-core/ecma/base/ecma-globals.h

```diff
@@ -227,56 +306,61 @@ typedef struct
 } ecma_getter_setter_pointers_t;
 
 /**
- * Description of ecma-property
+ * Property data.
  */
-typedef struct ecma_property_t
+typedef union
 {
-  /** Compressed pointer to next property */
-  mem_cpointer_t next_property_p;
+  ecma_value_t value; /**< value of a property */
+  ecma_getter_setter_pointers_t getter_setter_pair; /**< getter setter pair */
+} ecma_property_value_t;
 
-  /** Property's flags (ecma_property_flags_t) */
-  uint8_t flags;
+/**
+ * Property pair.
+ */
+typedef struct
+{
+  ecma_property_header_t header; /**< header of the property */
```

Comment:
Then, please, give a very short heads-up what will make use of the same header. Right now, what we have in this PR is but a linked list of pairs of properties, both of which have `type`, `value`, and `name_cp` fields organized in a layout efficient way. What's coming up next?